nlp/optimize/auglag.py

```diff
@@ -182,12 +182,12 @@ def least_squares_multipliers(self, x):
         Jred = ReducedJacobian(J, np.arange(m, dtype=np.int),
                                free_vars)
 
-        g = slack_model.grad(x)
+        g = slack_model.grad(x) - J.T * al_model.pi
 
         lsqr = LSQRSolver(Jred.T)
         lsqr.solve(g[free_vars], itnlim=lim)
         if lsqr.optimal:
-            al_model.pi = lsqr.x.copy()
+            al_model.pi += lsqr.x.copy()
```

Comment:
Yes. A couple of the CUTEst problems had difficulty converging when calculating the multipliers from scratch close to the optimal solution. This was fixed by computing the change in multipliers instead.